analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/VerifierBuilder.cs

```diff
@@ -58,11 +68,34 @@ private VerifierBuilder(VerifierBuilder original)
             new(this) { Paths = Paths.Concat(paths).ToImmutableArray() };
 
         public VerifierBuilder AddReferences(IEnumerable<MetadataReference> references) =>
-            new(this) { References = References.Concat(references).ToImmutableArray() };
+            references != null && references.Any()
+                ? new(this) { References = References.Concat(references).ToImmutableArray() }
+                : this;
+
+        public VerifierBuilder WithErrorBehavior(CompilationErrorBehavior errorBehavior) =>
+            new(this) { ErrorBehavior = errorBehavior };
+
+        public VerifierBuilder WithLanguageVersion(CS.LanguageVersion languageVersion) =>
+            WithOptions(ImmutableArray.Create<ParseOptions>(new CS.CSharpParseOptions(languageVersion)));
+
+        public VerifierBuilder WithLanguageVersion(VB.LanguageVersion languageVersion) =>
+            WithOptions(ImmutableArray.Create<ParseOptions>(new VB.VisualBasicParseOptions(languageVersion)));
+
+        public VerifierBuilder WithOnlyDiagnostics(params DiagnosticDescriptor[] onlyDiagnostics) =>
+            new(this) { OnlyDiagnostics = onlyDiagnostics.ToImmutableArray() };
 
         public VerifierBuilder WithOptions(ImmutableArray<ParseOptions> parseOptions) =>
             new(this) { ParseOptions = parseOptions };
 
+        public VerifierBuilder WithOutputKind(OutputKind outputKind) =>
+            new(this) { OutputKind = outputKind };
+
+        public VerifierBuilder WithSonarProjectConfigPath(string sonarProjectConfigPath) =>
+            new(this) { SonarProjectConfigPath = sonarProjectConfigPath };
+
+        public VerifierBuilder WithTopLevelStatements() =>
```

Comment:
I would remove this method:
- it's misleading (it should be ensure that the language versions contains C# 9 and no other previous version)
- can be called also for VB
- we can use directly `WithOutputKind(OutputKind.ConsoleApplication)` which is more clear about the behavior

Reply:
The next PR #5214 adds `.WithConcurrentAnalysis(false);` that needs to be always called with this.
I'll improve the C# 9 part.
I'm fine with ignoring the VB design flaw :)

Reply:
I'm still not sure about the value of this method. We can easily live without it

Reply:
We could live without it, and it will help us many times to save complicated copy/paste scaffolding that will always go together (considering the #5214 modification)